src/typescript/Scala.tmLanguage.ts

```diff
@@ -708,7 +708,7 @@ export const scalaTmLanguage: TmLanguage = {
           name: 'meta.package.scala'
         },
         {
-          match: `\\b(given)\\b\\s*(?:\\b(as)\\b|(${idUpper})|(${backQuotedId}|(?!//|/\\*)${plainid})?)`,
+          match: `\\b(given)\\b\\s*(?:\\b(as)\\b|(${idUpper})(?=.+\\bas\\b)|(${backQuotedId}(?=.+\\bas\\b)|(?!//|/\\*)${plainid}(?=.+\\bas\\b))?)`,
```

Comment:
`.+` can be too greedy, and make the regex match just because there is an `as` later on the line (e.g. `given foo = ???; given Bar as bar`, where `foo` is highlighted incorrectly).

I dont think we need `\b\s*`, we can just use `\s+`, which also allows us to remove the `\b` before the first `(as)`.

Reply:
I refactored it and added support for multiple given definitions in a single line. But we lost one of the complex cases with refinement types.